### PR TITLE
code improvement and formatting of rest stat client

### PR DESCRIPTION
**Pull request by ruks**
improve lucene search query by adding exact matching
remove unwanted stat method from host objects
define stat limit functions

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRestClientImpl.java [hunk 1]
```diff
@@ -1256,25 +1233,33 @@ public static OMElement buildOMElement(InputStream inputStream) throws Exception
...
         try {
             query = APIUsageStatisticsClientConstants.REQUEST_TIME + ": [" + RestClientUtil.getFloorDateAsLong(fromDate)
-                    + " TO " + RestClientUtil.getCeilingDateAsLong(toDate) + "]";
+                    + " TO " + RestClientUtil.getCeilingDateAsLong(toDate) + ']';
         } catch (ParseException e) {
             handleException("Error occurred while Error parsing date", e);
         }
 
+        //if My APIs stat, add constraint with API publisher
+        if (!providerName.equals(APIUsageStatisticsClientConstants.ALL_PROVIDERS)) {
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRestClientImpl.java]: We may do this otherway around contant.equals(variable) if provider will not be null then fine

File: components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRestClientImpl.java [hunk 2]
```diff
@@ -1408,11 +1392,19 @@ public static OMElement buildOMElement(InputStream inputStream) throws Exception
...
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRestClientImpl.java]: same here

File: components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRestClientImpl.java (hunk 1 above)
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRestClientImpl.java]: fixed in commit: https://github.com/wso2/carbon-apimgt/commit/9619da2271b49f37ef93456b6f8a8ac235f685ea

File: components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRestClientImpl.java (hunk 2 above)
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRestClientImpl.java]: fixed in commit: https://github.com/wso2/carbon-apimgt/commit/9619da2271b49f37ef93456b6f8a8ac235f685ea

File: components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRestClientImpl.java [hunk 3]
```diff
@@ -725,11 +724,19 @@ public static OMElement buildOMElement(InputStream inputStream) throws Exception
         //extending lucene query with time ranges
         try {
             query = APIUsageStatisticsClientConstants.REQUEST_TIME + ": [" + RestClientUtil.getFloorDateAsLong(fromDate)
-                    + " TO " + RestClientUtil.getCeilingDateAsLong(toDate) + "]";
+                    + " TO " + RestClientUtil.getCeilingDateAsLong(toDate) + ']';
         } catch (ParseException e) {
             handleException("Error occurred while Error parsing date", e);
         }
 
+        //if My APIs stat, add constraint with API publisher
```
Comment (nuwand) [inline components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRestClientImpl.java]: No need to maintain unwanted code.

Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRestClientImpl.java]: fixed in commit: https://github.com/wso2/carbon-apimgt/commit/c5071cbc7f3c39fed2489c288179e0fe93507544

**Pull request merged**